### DISCUSSION TITLE
Running a background with NextJS v12

### DISCUSSION THREAD

**Discussion by probablycorey** [vercel/next.js#39927] [Help] [answered]
In NextJS v11, I could easily spin up a long running background job very easily by adding the the next.config.js file with code like this https://gist.github.com/gregorym/81adc4971274ce19d7b77278e4d8cf33. 

But in v12 this doesn't work. The problem is the worker file (that includes node libs) is always compiled into the server and browser code. So when the browser runs it complains with this error `Module not found: Can't resolve 'fs'`

Is there any way to setup the next config so certain files are only compiled for the server and not for the browser?

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Could you try with:

```ts
webpack: (config, opts) => {
    console.log(opts.nextRuntime === 'nodejs');
    return config;
  },
```

Sort of:

```ts
webpack: (config, { buildId, dev, isServer, defaultLoaders, webpack, nextRuntime }) => {
    if (nextRuntime === 'nodejs') {
      return merge(config, {
        entry () {
          return config.entry().then((entry) => {
            return Object.assign({}, entry, { 'collection.worker': path.resolve(process.cwd(), 'workers/collection.worker.ts') })
          })
        }
      });
    } else {
      return config;
    }
  }
```

**Reply by probablycorey** [reply]
I ran `console.log("🤯", opts.nextRuntime, opts.nextRuntime === "nodejs")` below are the results.

```
🤯 nodejs true
🤯 edge false
🤯 undefined false
```

Using your `if (nextRuntime === 'nodejs')` solved the problem. Thanks @icyJoseph!
